mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1636,6 +1639,84 @@ private static ReadPreference getReadPreference(AssociationContext associationCo
 		return associationContext.getAssociationTypeContext().getOptionsContext().getUnique( ReadPreferenceOption.class );
 	}
 
+
+	@Override
+	public boolean supportsNamedPosition() {
+		return false;
+	}
+
+	/**
+	 * call stored procedure
+	 * @param storedProcedureName name of stored procedure.
+	 * @param params - array with values
+	 * @param tupleContext the tuple context
+	 *@see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java"> example</a>
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName, Object[] params,
+			TupleContext tupleContext) {
+		StringBuffer commandLine = new StringBuffer( storedProcedureName ).append( "(" );
+		for ( Object param : params ) {
+			if ( param instanceof String ) {
+				commandLine.append( '\'' ).append( param.toString() ).append( '\'' );
```

Comment:
If it's a String you can simply cast the value `(String) param`

Reply:
You right :+1:

Reply:
This one bothers me a bit. We don't escape the string before quoting it? It looks like a source for "NoSQL injection".

It would be nice if we could find an API in the MongoDB driver allowing to escape the strings properly.